src/lib.rs

```diff
@@ -1,744 +1,345 @@
-    IRFilterMaximumMinMagLinearMipPoint = 0x194,
-    IRFilterMaximumMinMagMipLinear = 0x195,
-    IRFilterMaximumAnisotropic = 0x1d5,
-}
-
-#[repr(u32)]
-#[derive(Copy, Clone)]
-pub enum IRTextureAddressMode {
-    IRTextureAddressModeWrap = 1,
-    IRTextureAddressModeMirror = 2,
-    IRTextureAddressModeClamp = 3,
-    IRTextureAddressModeBorder = 4,
-    IRTextureAddressModeMirrorOnce = 5,
-}
-
-#[repr(u32)]
-#[derive(Copy, Clone)]
-pub enum IRComparisonFunction {
-    IRComparisonFunctionNever = 1,
-    IRComparisonFunctionLess = 2,
-    IRComparisonFunctionEqual = 3,
-    IRComparisonFunctionLessEqual = 4,
-    IRComparisonFunctionGreater = 5,
-    IRComparisonFunctionNotEqual = 6,
-    IRComparisonFunctionGreaterEqual = 7,
-    IRComparisonFunctionAlways = 8,
+impl Drop for IRRootSignature {
+    fn drop(&mut self) {
+        unsafe { self.funcs.IRRootSignatureDestroy(self.me) }
+    }
 }
 
-#[repr(u32)]
-#[derive(Copy, Clone)]
-pub enum IRStaticBorderColor {
-    IRStaticBorderColorTransparentBlack = 0,
-    IRStaticBorderColorOpaqueBlack = 1,
-    IRStaticBorderColorOpaqueWhite = 2,
-}
+impl IRRootSignature {
+    pub fn create_from_descriptor(
+        compiler: &IRCompiler,
+        desc: &IRVersionedRootSignatureDescriptor,
+    ) -> Result<IRRootSignature, Box<dyn std::error::Error>> {
+        unsafe {
+            let mut error: *mut IRError = std::ptr::null_mut::<IRError>();
 
-#[repr(C)]
-#[derive(Copy, Clone)]
-pub struct IRStaticSamplerDescriptor {
-    pub filter: IRFilter,
-    pub address_u: IRTextureAddressMode,
-    pub address_v: IRTextureAddressMode,
-    pub address_w: IRTextureAddressMode,
-    pub mip_lod_bias: f32,
-    pub max_anisotropy: u32,
-    pub comparison_func: IRComparisonFunction,
-    pub border_color: IRStaticBorderColor,
-    pub min_lod: f32,
-    pub max_lod: f32,
-    pub shader_register: u32,
-    pub register_space: u32,
-    pub shader_visibility: IRShaderVisibility,
-}
+            let me = compiler
+                .funcs
+                .IRRootSignatureCreateFromDescriptor(desc, &mut error);
 
-#[repr(u32)]
-#[derive(Copy, Clone)]
-pub enum IRRootSignatureFlags {
-    IRRootSignatureFlagNone = 0,
-    IRRootSignatureFlagAllowInputAssemblerInputLayout = 0x1,
-    IRRootSignatureFlagDenyVertexShaderRootAccess = 0x2,
-    IRRootSignatureFlagDenyHullShaderRootAccess = 0x4,
-    IRRootSignatureFlagDenyDomainShaderRootAccess = 0x8,
-    IRRootSignatureFlagDenyGeometryShaderRootAccess = 0x10,
-    IRRootSignatureFlagDenyPixelShaderRootAccess = 0x20,
-    IRRootSignatureFlagAllowStreamOutput = 0x40,
-    IRRootSignatureFlagLocalRootSignature = 0x80,
-    IRRootSignatureFlagDenyAmplificationShaderRootAccess = 0x100,
-    IRRootSignatureFlagDenyMeshShaderRootAccess = 0x200,
-    IRRootSignatureFlagCBVSRVUAVHeapDirectlyIndexed = 0x400,
-    IRRootSignatureFlagSamplerHeapDirectlyIndexed = 0x800,
-}
+            Ok(Self {
+                funcs: compiler.funcs.clone(),
+                me,
```

Comment:
Nit: use `rust-analyzer` to sort these.

Reply:
`rust-analyzer` did not catch this, I'll fix it manually in a followup

Reply:
Let's do it straight away.